### PR TITLE
Add simplify rsqrt pass

### PR DESCRIPTION
**Pull request by kahmed10**

### CODE REVIEW

File: src/simplify_algebra.cpp [hunk 1]
```diff
@@ -391,6 +392,23 @@ struct find_sub_const
     }
 };
 
+struct find_rsqrt
+{
+    auto matcher() const
+    {
+        return match::name("recip")(
+            match::args(match::name("sqrt")(match::args(match::any().bind("x")))));
```
Comment (scxiao) [inline src/simplify_algebra.cpp:400]: Should we consider output of sqrt used only once?

Comment (kahmed10) [reply src/simplify_algebra.cpp:401]: fixed

File: src/simplify_algebra.cpp [hunk 2]
```diff
@@ -396,8 +396,8 @@ struct find_rsqrt
 {
     auto matcher() const
     {
-        return match::name("recip")(
-            match::args(match::name("sqrt")(match::args(match::any().bind("x")))));
+        return match::name("recip")(match::args(
+            match::name("sqrt")(match::used_once(), match::args(match::any().bind("x")))));
```
Comment (scxiao) [inline src/simplify_algebra.cpp:400]: Can you add a unit test to verify this?

Comment (kahmed10) [reply src/simplify_algebra.cpp:400]: added

**Pull request merged**